app/src/main/java/com/mk/tasky/agenda/data/AgendaRepositoryImpl.kt

```diff
@@ -62,6 +67,40 @@ class AgendaRepositoryImpl(
         return deleteReminderByIdRemotely(id)
     }
 
+    override fun getAgenda(date: LocalDateTime, forceRemote: Boolean): Flow<Agenda> {
+        // TODO: Update with Tasks and Events
+        return flow {
+            val localReminders = getRemindersForDate(date.toLocalDate()).toMutableList()
+            emit(Agenda(reminders = localReminders, tasks = emptyList(), events = emptyList()))
+            if (forceRemote) {
+                getAgendaRemotely(date).onSuccess { response ->
+                    response.reminders.forEach {
+                        dao.insertReminder(it.toDomain().toEntity())
```

Comment:
Performance improvement:
```
response.reminders.map {
    launch { dao.insertReminder(it.toDomain().toEntity()) }
}.forEach { it.join() }
```

Right now, you insert each reminder one after another. With the above code all reminders are added at the same time in parallel which will lead to roughly 10x faster code when inserting 10 reminders, 100x faster code when inserting 100 reminders and so on 😄 
Likely not very noticeable since inserting is super quick, but as soon as that's a network call that wouldn't be so nice anymore